library/src/main/scala/scala/scalajs/js/JSApp.scala

```diff
@@ -2,13 +2,28 @@ package scala.scalajs.js
 
 /** Base class for top-level, entry point main objects.
  *
- *  [[JSApp]] is typically used to mark the entry point of a Scala.js
- *  application. As such, the sbt plugin also recognizes top-level objects
- *  extending [[JSApp]]. It allows to run their [[main]] method with `sbt run`.
+ *  In Scala.js 0.6.x, an top-level object had to extend `js.JSApp` to be
+ *  recognized by the sbt plugin as a "main" object, to be executed with `run`.
+ *  Starting with Scala.js 1.0.0, any object with a standard main method of the
+ *  form
+ *  {{{
+ *  def main(args: Array[String]): Unit = ???
+ *  }}}
+ *  will be recognized by the sbt plugin, just like for a JVM project.
  *
- *  To execute the [[main]] method immediately when your Scala.js file is
- *  loaded, use the `scalaJSUseMainModuleInitializer` setting in the sbt plugin.
+ *  [[JSApp]] is therefore deprecated, and should not be used anymore.
  */
+@deprecated(
+    "Extending js.JSApp is not necessary anymore for an object to be " +
+    "recognized by the Scala.js sbt plugin. " +
+    "Use a normal object with a `def main(args: Array[String]): Unit` " +
+    "instead, which will also be cross-platform. " +
+    "Note that js.JSApp objects and their main() method are not " +
+    "automatically exported to JavaScript either; explicitly export your " +
+    "object if you relied on this behavior.",
+    "1.0.0")
```

Comment:
Hum, so you were thinking that we would remove `@ScalaJSDefined` altogether before 1.0.0 final? I thought it was pretty much a given that it was going to stay there in limbo in 1.x. `js.JSApp` is in the same situation.

Now, it's an interesting thought to deprecate them in the Milestones and remove them in the RCs. But that means users cannot jump straight from 0.6.latest to 1.0.0 final. They need to take a corner through the latest 1.0.0-Mx.

Unless we're bad-ass and we deprecate `@ScalaJSDefined` in 0.6.18, with a message saying that users should activate `-P:scalajs:sjsDefinedByDefault` and remove the annotations.

If we're willing to do that for `@ScalaJSDefined`, then we should indeed find an equally good solution for `js.JSApp`. I'd be very happy not to it linger in the library forever!